eras/conway/impl/cddl-files/conway.cddl

```diff
@@ -48,7 +48,12 @@ next_major_protocol_version = 10
 
 major_protocol_version = 1..next_major_protocol_version
 
-protocol_version = (major_protocol_version, uint)
+protocol_version =
+  ( major_protocol_version
+  , minor_protocol_version : uint
+  )
+
+protocol_version_ = [ protocol_version ]
```

Comment:
Wait! This suggestion opened my eyes.

Actually I never dealt with an `header_body`, so actually I have no idea of any implications in that context...

The idea of opening this PR cames from the fact that `cardano-node v8.8-pre` accepts transactions encoded with this CBOR:

```
83                          # array(3)
   01                       # unsigned(1)
   F6                       # primitive(22)
   82                       # array(2)
      09                    # unsigned(9)
      00                    # unsigned(0)
```
(if I understood correctly, a CBOR reflecting following syntax)
```
protocol_version = (major_protocol_version, uint)

hard_fork_initiation_action = (1, gov_action_id / null, protocol_version)
```

while if we try to subit a transaction encoded with following CBOR, we get a deserialization error.

```
83                          # array(3)
   01                       # unsigned(1)
   F6                       # primitive(22)
   81                       # array(1)
      82                    # array(2)
         09                 # unsigned(9)
         00                 # unsigned(0)
```
(if I understood correctly, a CBOR reflecting following syntax - current syntax)
```
protocol_version = (major_protocol_version, uint)

hard_fork_initiation_action = (1, gov_action_id / null, [ protocol_version ])
```

Could it be the problem is just a misalignment between this document and `cardano-node`?

Reply:
Thank you for your explaination and please ignore my previous question, I'm still learning about CBOR and CDDL...

43b682412 should address your suggestion.

Last, about your proposal:

> If you'd like to fix this for all previous eras that would be great too. This would make it consistent across eras

Actually I tried, but after reading #3559 and #3762 I was no longer sure about the change.

If you can confirm that it is correct to change `protocol_version` and how it is referenced in `header_body` like in this PR for all the eras, I'll be happy to do that.

Thank you

Reply:
Oh, yeah. If you'd like you could apply the same change you did here to the Babbage era, but earlier eras had a different encoding in the block body header. They used [grouping](https://github.com/IntersectMBO/cardano-ledger/blob/bc2fdce909fe27e8af93c5c308fda9b16a861bc3/libs/cardano-protocol-tpraos/src/Cardano/Protocol/TPraos/BHeader.hs#L250), which essentially unwraps and inlines the elements of the array. So let's leave pre-Babbage eras as is.

Reply:
f584a37de should address this.

Thank yuo for your patience.